### PR TITLE
Improve requested claims consent experience on application claim configuration update

### PR DESCRIPTION
**Pull request by GANGANI**
## Purpose
With the current implementation, it will not ask the consent for the requested claims which were added after the initial consent request. Hense with the PR changes, will resolve that issue and will improve the requested claims consent experience on application claim configuration update.
**Resolves:** wso2/product-is#9009
**Related PR:** https://github.com/wso2/carbon-consent-management/pull/180

## Approach
Introduced the "IS_CONSENTED" column to the "CM_SP_PURPOSE_PII_CAT_ASSOC" table to store the consent of the user for the particular application. With this PR changes, it will process the consent and get the claims with consent and without consent and store them accordingly and will derive the claims that have not been processed to give the consent.

### When should this PR be merged
This PR should be merged after merging https://github.com/wso2/carbon-consent-management/pull/180

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java [hunk 1]
```diff
@@ -246,7 +248,6 @@ private String getSPTenantDomain(ServiceProvider serviceProvider) {
         Set<String> consentClaims = getUniqueLocalClaims(requestedClaims, mandatoryClaims);
```
Comment (IsuraD) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java:248]: Do we need to add mandatoryClaims now? We shouldn't add ?

Comment (GANGANI) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java]: Yes. It's not required to add. Fixed https://github.com/wso2/carbon-identity-framework/pull/3413/commits/21d2c876c2af1be806c780d1099daeb16466692b

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java [hunk 2]
```diff
@@ -214,11 +216,14 @@ private boolean hasConsentForRequiredClaims(ConsentClaimsData consentClaimsData)
     private void removeClaimsWithoutConsent(AuthenticationContext context, ConsentClaimsData consentClaimsData)
             throws PostAuthenticationFailedException {
 
-        List<ClaimMetaData> claimsWithConsent = consentClaimsData.getClaimsWithConsent();
-        List<String> claimsURIsWithConsent = getClaimsFromMetaData(claimsWithConsent);
-        Set<String> claimsWithoutConsent = getClaimsWithoutConsent(claimsURIsWithConsent, context);
+        List<ClaimMetaData> claimsWithConsentAndRequestedConsent = consentClaimsData.getClaimsWithConsent();
```
Comment (IsuraD) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java:219]: approvedAndNewlyRequestedConsents

Comment (GANGANI) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java]: Fixed https://github.com/wso2/carbon-identity-framework/pull/3413/commits/cd7b7f24a73595e087817483816749ec89ae0444

**Pull request merged**